python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -99,132 +84,134 @@ def __init__(self, system, restraint_dict, type='Boresch'):
                     raise ValueError(
                         f"restraint_dict['force_constants']['{key}'] must be of type "
                         f"'BioSimSpace.Types.Energy'/'BioSimSpace.Types.Angle^2'")
-
-            self._content = restraint_dict
-
         else:
             raise NotImplementedError(f'Restraint type {type} not implemented '
                                       f'yet. Only boresch restraint is supported.')
 
+        self._content = restraint_dict
+        self.update_system(system)
+
+    def update_system(self, system):
+        '''Update the system object.'''
+        if not isinstance(system, _System):
+            raise TypeError("'system' must be of type 'BioSimSpace._SireWrappers.System'")
+        else:
+            if self.rest_type == 'boresch':
+                # Check if the ligand atoms are decoupled.
+                # Find the decoupled molecule, assume that only one can be
+                # decoupled.
+                (decoupled_mol,) = system.getDecoupledMolecules()
+                for key in ['l1', 'l2', 'l3']:
+                    atom = self._content['anchor_points'][key]
+                    if not atom._sire_object.molecule().number() == decoupled_mol._sire_object.number():
+                        raise ValueError(
+                            f'The ligand atom {key} is not from decoupled moleucle.')
+
+            # Store a copy of solvated system.
+            self._system = system.copy()
+
     def toString(self, engine='Gromacs'):
         '''The method for convert the restraint to a format that could be used
```

Comment:
I realise you're in a rush, but if you have time later, maybe make the docstrings more complete in general (e.g. include parameters)?

Reply:
Sorry, I will add them now